File: TOKI/Problems/TROC #29/b.pas
var n, x, y, i, odd, even: Int64;

begin

  odd := 0;
  even := 0;

  readLn(n);  

  for i := 1 to n do
  begin
    readLn(x, y);

    if (x+y) mod 2 = 1 then
        odd := odd + 1
    else
        even := even + 1;
  end;

  if (odd >= 1) and (even >= 1) then
    writeLn('YES')
  else 
    writeLn('NO');

end.

File: TOKI/Problems/TROC #29/b.go
package main

import (
	"fmt"
)

func main() {
	var n, x, y, i int64
	odd := 0
	even := 0

	fmt.Scanf("%d\n", &n)

	for i = 0; i < n; i++ {
		fmt.Scanf("%d %d\n", &x, &y)

		if (x+y)%2 == 1 {
			odd++
		} else {
			even++
		}
	}

	if odd >= 1 && even >= 1 {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}
}
